src/pyodide.py

```diff
@@ -39,4 +39,22 @@ def eval_code(code, ns):
         return None
 
 
-__all__ = ['open_url', 'eval_code']
+def find_imports(code):
+    """
+    Finds the imports in a string of code and returns a list of their package
+    names.
+    """
+    mod = ast.parse(code)
+    imports = set()
+    for node in ast.walk(mod):
+        if isinstance(node, ast.Import):
+            for name in node.names:
+                name = name.name
+                imports.add(name.split('.')[0])
+        elif isinstance(node, ast.ImportFrom):
+            name = node.module
+            imports.add(name.split('.')[0])
+    return list(imports)
```

Comment:
Maybe it would be worth adding some test for this? For instance,

**conftest.py**
```py

@pytest.fixture
def noselenium():
    """A fixture to allow running tests that don't require selenium"""
    return 'noselenium'
```

**test/test_package_loading.py**
```py
def test_import_determination(noselenium):
   imports_list = find_imports(
        """
        import numpy as np
        np.arange(10)
        """)
   assert imports_list == ['numpy']
   # etc
```

then create yet another CI test job with,
```sh
pytest -v tests -k noselenium
```
to run tests that don't need selenium.

It's a bit awkward to have this `noselenium` fixture, I'm not sure there is a better way.  I think sooner or later we would need some mechanisms to run pure python tests anyway..

Reply:
Also maybe add a test to check that this works, something like,
```py
def test_load_packages_multiple(selenium_standalone):
    selenium = selenium_standalone
    selenium.run_async("""
         import pyparsing
         """)
```
where `run_async` is the appropriately defined method? Or is it problematic to do that because it's async?